### PR TITLE
Geographical representation of hvdc

### PR DESCRIPTION
**Pull request by anistouri**

### CODE REVIEW

**Comment by etiennehomer** [comment]
You need to update the hvdc collection when the user changes the current node and when the network-map-tab get the updatedSubstationsToSend.
See methods loadMapEquipments(), loadAllGeoData() and updateMapEquipmentsAndGeoData()

File: src/components/menus/branch-menu.js [hunk 1]
```diff
@@ -74,22 +74,32 @@ const withBranchMenu =
                 equipmentType === equipments.lines ? 'Line' : '2WTransformer'
```
Comment (jonenst) [inline src/components/menus/branch-menu.js:74]: to update ?

File: src/components/study-container.js [hunk 2]
```diff
@@ -635,7 +635,11 @@ export function StudyContainer({ view, onChangeTab }) {
                     displayNetworkLoadingFailMessage,
                     dispatch,
                     {
-                        equipments: [equipments.lines, equipments.substations],
+                        equipments: [
+                            equipments.lines,
+                            equipments.substations,
+                            equipments.hvdcLines,
```
Comment (etiennehomer) [inline src/components/study-container.js:641]: You can remove this prefetch

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/network/network-map.js [hunk 3]
```diff
@@ -94,10 +94,17 @@ const NetworkMap = (props) => {
 
     const readyToDisplayLines =
         readyToDisplay &&
-        props.mapEquipments.lines &&
+        (props.mapEquipments?.lines || props.mapEquipments?.hvdcLines) &&
```
Comment (etiennehomer) [inline src/components/network/network-map.js:97]: I'm not sure of this condition. hvdcLines may be downloaded quicker. Do we want to make a deck.gl render with only hvdc lines ?

**Comment by etiennehomer** [comment]
Tests OK

File: src/components/menus/base-equipment-menu.js [hunk 4]
```diff
@@ -152,6 +152,7 @@ const BaseEquipmentMenu = ({
     const equipmentsNotDeletable = [
         equipments.lccConverterStations,
         equipments.vscConverterStations,
+        equipments.hvdcLines,
```
Comment (etiennehomer) [inline src/components/menus/base-equipment-menu.js:155]: Do the PO explictly want to wait before making the "hvdc delete from the map" available ?

File: src/components/menus/branch-menu.js [hunk 5]
```diff
@@ -75,21 +75,32 @@ const withBranchMenu =
             );
```
Comment (jonenst) [inline src/components/menus/branch-menu.js:75]: need to take hvdc-line into account

Comment (jonenst) [reply src/components/menus/branch-menu.js:75]: the menu is empty for hvdc lines ?

File: src/components/menus/base-equipment-menu.js (hunk 4 above)
Comment (etiennehomer) [reply src/components/menus/base-equipment-menu.js:155]: Discussed: we keep it hidden for the moment

**Review by etiennehomer — APPROVED** [review]

File: src/components/menus/branch-menu.js (hunk 5 above)
Comment (anistouri) [reply src/components/menus/branch-menu.js:75]: For now, we just need "view in spreadsheet" in hvdc lines menu

File: src/components/menus/branch-menu.js (hunk 1 above)
Comment (anistouri) [reply src/components/menus/branch-menu.js]: this breaks translation

File: src/components/network/map-equipments.js [hunk 6]
```diff
@@ -61,6 +69,22 @@ export default class MapEquipments {
                     );
                 }
             });
+        fetchMapHvdcLines(studyUuid, currentNodeUuid, undefined, false)
+            .then((val) => {
+                this.dispatch(
+                    mapEquipmentsCreated(this, undefined, undefined, val)
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/components/network/map-equipments.js:80]: Done

File: src/components/menus/branch-menu.js [hunk 7]
```diff
@@ -70,26 +70,35 @@ const withBranchMenu =
         const [branch, setBranch] = useState(null);
 
         const getTranslationKey = (key) => {
-            return key.concat(
-                equipmentType === equipments.lines ? 'Line' : '2WTransformer'
-            );
+            return key.concat(getEquipmentPath(equipmentType));
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/components/menus/branch-menu.js]: Done

File: src/components/network-map-tab.js [hunk 8]
```diff
@@ -617,13 +618,28 @@ export const NetworkMapTab = ({
...
+                    currentNodeRef.current?.id
+                ) {
+                    mapEquipments.updateHvdcLines(
+                        mapEquipments.checkAndGetValues(values),
+                        isFullReload
+                    );
+                    updatedLinesArray.push(values);
+                }
+            });
+            setUpdatedLines(updatedLinesArray);
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/components/network-map-tab.js]: Done

**Review by jonenst — APPROVED** [review]

**Pull request merged**